dialogue-target/src/main/java/com/palantir/dialogue/BindEndpoint.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue;
+
+public interface BindEndpoint {
+    EndpointChannel bindEndpoint(Endpoint endpoint);
+}
```

Comment:
this is public because `DefaultClients` lives in `:dialogue-serde` and doesn't depend on `:dialogue-core`, but needs to be able to do a bind nicely.

Reply:
Could we write things through the runtime object? I'd like the runtime to be capable of taking a channel and endpoint, and returning an EndpointChannel.

Reply:
Is this kinda what you were imagining? https://github.com/palantir/dialogue/pull/764/files#r426760705

Reply:
I think moving this particular interface anywhere would change our cross project dependencies (e.g. if we made `:core` depend on `:serde` or vice versa)